core/src/main/java/io/apigee/trireme/core/modules/crypto/HashImpl.java

```diff
@@ -83,15 +73,18 @@ private void initializeHash(Context cx, Object[] args, Function ctorObj)
     {
         String nodeAlgorithm = stringArg(args, 0);
 
-        String jceAlgorithm = MD_ALGORITHMS.get(nodeAlgorithm);
-        if (jceAlgorithm == null) {
-            jceAlgorithm = nodeAlgorithm;
+        HashAlgorithms.Algorithm alg = HashAlgorithms.get().get(nodeAlgorithm);
+        if (alg == null) {
+            alg = HashAlgorithms.get().getByJavaHashName(nodeAlgorithm);
+        }
+        if (alg == null) {
+            throw Utils.makeError(cx, ctorObj, "Digest method not supported");
```

Comment:
Would it make sense to have the digest at the end of the error?  For example: `Digest method not supported: fake-algo`